### ISSUE TITLE
Intuition behind separating recurrent_kernel in LSTM

### ISSUE THREAD

**Issue by felixhao28** [keras-team/keras#9773]
Hi, does anyone understand the intuition behind different default initializers for 'kernel' and 'recurrent kernel' in Keras LSTM implementation?

I am currently digging an issue where my Keras model has a consistent ~5% better accuracy than my TensorFlow counterpart. The only difference I can see is that by default [Keras uses 'orthogonal' for recurrent_initializer](https://github.com/keras-team/keras/blob/master/keras/layers/recurrent.py#L1746) and [TensorFlow uses 'glorot_uniform' for both kernels](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/rnn_cell_impl.py#L763). But I really doubt the initialization method will make a huge difference like 5%.
